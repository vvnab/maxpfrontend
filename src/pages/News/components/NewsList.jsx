import React from 'react';
import PropTypes from 'prop-types';

const NewsList = ({news}) => {
  return !!news.length && <div>
                            <ol>
                              { news.map(i => (
                                  <li key={ i.id }>
                                    <h4>{ i.title }</h4>
                                    <p>
                                      { i.text }
                                    </p>
                                  </li>
                                )) }
                            </ol>
                            <br/>
                            <strong> { `всего новостей: ${news.length}` } </strong>
                          </div>
}

NewsList.propTypes = {
  news: PropTypes.array.isRequired
}

export default NewsList;