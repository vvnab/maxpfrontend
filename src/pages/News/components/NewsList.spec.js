import React from 'React'
import { shallow } from 'enzyme'
import NewsList from './NewsList'

describe('NewsList', () => {
  const props = {
    news: [{
      id: 1,
      title: 'title-1',
      text: 'text-1'
    }, {
      id: 2,
      title: 'title-2',
      text: 'text-2'
    }]
  }
  describe('NewsList render', () => {
    const nextProps = {
      ...props
    }
    const newsList = shallow(<NewsList {...nextProps} />);
    it('renders properly', () => {
      expect(newsList).toMatchSnapshot();
    })
    it('ok', () => {
      expect(newsList.find('ol')).toHaveLength(1);
      expect(newsList.find('li')).toHaveLength(nextProps.news.length);
      expect(newsList.find('li > h4')).toHaveLength(nextProps.news.length);
      expect(
        newsList
          .find('li > h4')
          .first()
          .text()
      ).toEqual(nextProps.news[0].title);
      expect(
        newsList
          .find('li > p')
          .first()
          .text()
      ).toEqual(nextProps.news[0].text);
      expect(
        newsList
          .find('strong')
          .text()
      ).toEqual(`всего новостей: ${nextProps.news.length}`);
    })
  })

  describe('NewsList empty', () => {
    const nextProps = {
      ...props,
      news: []
    }
    const newsList = shallow(<NewsList {...nextProps} />);
    it('renders properly', () => {
      expect(newsList).toMatchSnapshot();
    })
    it('ok', () => {
      expect(newsList.find('ol')).toHaveLength(0);
    })
  })
})