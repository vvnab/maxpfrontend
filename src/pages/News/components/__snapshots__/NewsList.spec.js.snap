// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsList NewsList empty renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsList
    news={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`NewsList NewsList render renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsList
    news={
        Array [
            Object {
              "id": 1,
              "text": "text-1",
              "title": "title-1",
            },
            Object {
              "id": 2,
              "text": "text-2",
              "title": "title-2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ol>
          <li>
                    <h4>
                              title-1
                    </h4>
                    <p>
                              text-1
                    </p>
          </li>
          <li>
                    <h4>
                              title-2
                    </h4>
                    <p>
                              text-2
                    </p>
          </li>
</ol>,
        <br />,
        <strong>
          всего новостей: 2
</strong>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <h4>
                            title-1
              </h4>
              <p>
                            text-1
              </p>
</li>,
            <li>
              <h4>
                            title-2
              </h4>
              <p>
                            text-2
              </p>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  title-1
</h4>,
                <p>
                  text-1
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "title-1",
                },
                "ref": null,
                "rendered": "title-1",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "text-1",
                },
                "ref": null,
                "rendered": "text-1",
                "type": "p",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  title-2
</h4>,
                <p>
                  text-2
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "title-2",
                },
                "ref": null,
                "rendered": "title-2",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "text-2",
                },
                "ref": null,
                "rendered": "text-2",
                "type": "p",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ol",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "всего новостей: 2",
        },
        "ref": null,
        "rendered": "всего новостей: 2",
        "type": "strong",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ol>
            <li>
                        <h4>
                                    title-1
                        </h4>
                        <p>
                                    text-1
                        </p>
            </li>
            <li>
                        <h4>
                                    title-2
                        </h4>
                        <p>
                                    text-2
                        </p>
            </li>
</ol>,
          <br />,
          <strong>
            всего новостей: 2
</strong>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <h4>
                                title-1
                </h4>
                <p>
                                text-1
                </p>
</li>,
              <li>
                <h4>
                                title-2
                </h4>
                <p>
                                text-2
                </p>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    title-1
</h4>,
                  <p>
                    text-1
</p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "title-1",
                  },
                  "ref": null,
                  "rendered": "title-1",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "text-1",
                  },
                  "ref": null,
                  "rendered": "text-1",
                  "type": "p",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    title-2
</h4>,
                  <p>
                    text-2
</p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "title-2",
                  },
                  "ref": null,
                  "rendered": "title-2",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "text-2",
                  },
                  "ref": null,
                  "rendered": "text-2",
                  "type": "p",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ol",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "всего новостей: 2",
          },
          "ref": null,
          "rendered": "всего новостей: 2",
          "type": "strong",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
